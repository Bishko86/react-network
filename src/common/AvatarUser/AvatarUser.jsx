import React from 'react';

const AvatarUser = ({ radius, width, height, url, }) => {
    return (
        <>
            <img style={{ width: width, height: height, borderRadius: radius }} src={url} alt="userphoto" />
        </>
    );
}
export default AvatarUser;