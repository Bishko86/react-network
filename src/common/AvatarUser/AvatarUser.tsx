import React, { FC } from 'react';

type PropsType = {
    width: number
    height: number
    radius: string
    url: string
}

const AvatarUser: FC<PropsType> = ({ radius, width, height, url, }): JSX.Element => {
    return (
        <>
            <img style={{ width: width, height: height, borderRadius: radius }} src={url} alt="userphoto" />
        </>
    );
}
export default AvatarUser;