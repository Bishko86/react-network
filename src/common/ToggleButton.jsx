import React from 'react';
import style from './../components/Users/Users.module.css';

const ToggleButton = ({ followingInProgress, userId, value, ...props }) => {
  return (<>
    <button disabled={followingInProgress.some(id => id === userId)} onClick={() => { props.followUnfollow(userId) }}
      className={style.followBtn}>{value}</button>
  </>
  )

}

export default ToggleButton;