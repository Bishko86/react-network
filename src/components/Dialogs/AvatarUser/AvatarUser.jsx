import React from 'react';
import style from './AvatarUser.module.css';

const AvatarUser = (props) => {

    return (
        <div className={style.avatar}>
            <img src={props.url} alt="" />
        </div>
    );
}
export default AvatarUser;