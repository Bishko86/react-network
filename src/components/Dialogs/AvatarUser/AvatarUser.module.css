.avatar img {
    width: 30px;
    height: 30px;
    border-radius: 50%;
    margin: 0 10px;
}
/* .avatar {
    position: relative;
    width: 30px;
    height: 30px;
    background-color: antiquewhite;
    border-radius: 50%;

} */