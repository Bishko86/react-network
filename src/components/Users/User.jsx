import React from 'react';
import { NavLink } from 'react-router-dom';
import ToggleButton from '../../common/ToggleButton';
import AvatarUser from '../../common/AvatarUser/AvatarUser';
import style from './Users.module.css';

const User = ({ user, ...props }) => {
  return (
    <div className={style.userBlock} >
      <div className={style.avatar}>
        <NavLink to={`profile/${user.id}`}>
          <AvatarUser url={user.photos.large || 'https://st2.depositphotos.com/2703645/5669/v/950/depositphotos_56695985-stock-illustration-male-avatar.jpg'} radius={'10px'} width={50} heigth={50} />
        </NavLink>
        {user.followed ?

          <ToggleButton followingInProgress={props.followingInProgress}
            followUnfollow={props.unFollow} userId={user.id} value='Unfollow'
          />
          :
          <ToggleButton followingInProgress={props.followingInProgress}
            followUnfollow={props.follow} userId={user.id} value='Follow'
          />

        }

      </div>
      <div className={style.userInfo}>
        <div>
          <h3>{user.name}</h3>
          <p>{user.status || null}
            {user.id}</p>
        </div>
      </div>
    </div>

  )
}

export default User;