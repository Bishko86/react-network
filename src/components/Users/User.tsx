import React from 'react';
import { NavLink } from 'react-router-dom';
import ToggleButton from '../../common/ToggleButton';
import AvatarUser from '../../common/AvatarUser/AvatarUser';
import style from './Users.module.css';
import { ItemsUserType } from './../../types/types'

type PropsType = {
  followingInProgress: Array<number>
  user: ItemsUserType
  unFollow: (id: number) => void
  follow: (id: number) => void
}
const User = ({ user, followingInProgress, unFollow, follow }: PropsType) => {
  // debugger
  return (
    <div className={style.userBlock} >
      <div className={style.avatar}>
        <NavLink to={`profile/${user.id}`}>
          <AvatarUser url={user.photos.large || 'https://st2.depositphotos.com/2703645/5669/v/950/depositphotos_56695985-stock-illustration-male-avatar.jpg'} radius={'10px'} width={50} height={50} />
        </NavLink>
        {user.followed ?

          <ToggleButton followingInProgress={followingInProgress}
            followUnfollow={unFollow} userId={user.id} value='Unfollow'
          />
          :
          <ToggleButton followingInProgress={followingInProgress}
            followUnfollow={follow} userId={user.id} value='Follow'
          />

        }

      </div>
      <div className={style.userInfo}>
        <div>
          <h3>{user.name}</h3>
          <p>{user.status || null}
            {user.id}</p>
        </div>
      </div>
    </div>

  )
}

export default User;