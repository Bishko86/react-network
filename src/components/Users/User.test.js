import React from 'react';
import renderer from 'react-test-renderer';
import { create } from "react-test-renderer";
import { NavLink } from 'react-router-dom';
import User from './User';
import AvatarUser from '../Dialogs/AvatarUser/AvatarUser';
import style from './Users.module.css';
import ToggleButton from './../../common/ToggleButton';
import { BrowserRouter } from 'react-router-dom';
const props = {
  user: {
    name: "Juls2201",
    id: 16690,
    uniqueUrlName: null,
    photos: {
      small: null,
      large: null
    },
    status: null,
    followed: false
  },
  followingInProgress: [],
  unFollow() { this.user.followed = false },
  follow() { props.user.followed = true }
}


// describe('Test of User Component', () => {
//   test('Testing follow / unfollow button', () => {
//     const component = renderer.create(<User  {...props} />);
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();

//     const instance = component.root;
//     const button = instance.findByType("button");
//     button.props.onClick()
//     tree = component.toJSON();
//     expect(tree).toMatchSnapshot();

//   });
// }
// )