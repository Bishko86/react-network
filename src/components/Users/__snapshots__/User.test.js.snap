// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of User Component Testing follow / unfollow button 1`] = `
<div
  className="userBlock"
>
  <div
    className="avatar"
  >
    <div
      className="avatar"
    >
      <img
        alt=""
        src="https://st2.depositphotos.com/2703645/5669/v/950/depositphotos_56695985-stock-illustration-male-avatar.jpg"
      />
    </div>
    <button
      className="followBtn"
      disabled={false}
      onClick={[Function]}
    >
      Follow
    </button>
  </div>
  <div
    className="userInfo"
  >
    <div>
      <h3>
        Juls2201
      </h3>
      <p>
        16690
      </p>
    </div>
  </div>
</div>
`;

exports[`Test of User Component Testing follow / unfollow button 2`] = `
<div
  className="userBlock"
>
  <div
    className="avatar"
  >
    <div
      className="avatar"
    >
      <img
        alt=""
        src="https://st2.depositphotos.com/2703645/5669/v/950/depositphotos_56695985-stock-illustration-male-avatar.jpg"
      />
    </div>
    <button
      className="followBtn"
      disabled={false}
      onClick={[Function]}
    >
      Follow
    </button>
  </div>
  <div
    className="userInfo"
  >
    <div>
      <h3>
        Juls2201
      </h3>
      <p>
        16690
      </p>
    </div>
  </div>
</div>
`;
